awscli/customizations/s3/results.py

```diff
@@ -246,8 +246,11 @@ def _get_ongoing_dict_key(self, result):
                 'Any result using _get_ongoing_dict_key must subclass from '
                 'BaseResult. Provided result is of type: %s' % type(result)
             )
-        return ':'.join(
-            str(el) for el in [result.transfer_type, result.src, result.dest])
+        key_parts = []
+        for result_property in [result.transfer_type, result.src, result.dest]:
+            if result_property is not None:
```

Comment:
I think the ongoing key will change if this `None` check is used. Might be worth to double check, but from what I can tell it is fine to change as the key is only used to get the a value out of dictionary. If the key was used for anything else, there may be issues.